### PR TITLE
Strip minuses from the hash names

### PR DESCRIPTION
**Pull request by l29ah**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

So sha256 and probably other authentihashes work
Only tested with a sha256-hashed PE

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Comment by l29ah** [comment]
Damn `hub`.

**Comment by radare** [comment]
which hashes have dashes?

**Comment by radare** [comment]
can you provide a test or example?

**Comment by l29ah** [comment]
I've observed "sha-256".
I don't have a way to produce signed binaries, and the one i've checked it on is proprietary.

File: libr/bin/format/pe/pe.c [hunk 1]
```diff
@@ -854,8 +854,8 @@ const char* PE_(bin_pe_compute_authentihash)(struct PE_(r_bin_pe_obj_t)* bin) {
 
 	const char *hashtype = bin->spcinfo->messageDigest.digestAlgorithm.algorithm->string;
 	ut64 algobit = r_hash_name_to_bits (hashtype);
```
Comment (radare) [inline libr/bin/format/pe/pe.c]: i think you must pass the filtered string in here, like this: r_str_replace_char (hashtype, '-', 0); (hashtype should be char* and strdupped from that .string

File: libr/hash/hash.c [hunk 2]
```diff
@@ -302,8 +302,12 @@ R_API ut64 r_hash_name_to_bits(const char *name) {
 
 	do {
 		/* Eat everything up to the comma */
-		for (i = 0; *ptr && *ptr != ',' && i < sizeof (tmp) - 1; i++) {
-			tmp[i] = tolower ((ut8)*ptr++);
+		for (i = 0; *ptr && *ptr != ',' && i < sizeof (tmp) - 1;) {
```
Comment (radare) [inline libr/hash/hash.c]: remove this logic, this should be implemented using RStr apis and not in here

**Pull request merged**